addon/adapters/pouch.js

```diff
@@ -26,13 +26,19 @@ export default DS.RESTAdapter.extend({
   shouldReloadRecord: function () { return false; },
   shouldBackgroundReloadRecord: function () { return false; },
 
-  _startChangesToStoreListener: on('init', function () {
-    this.changes = this.get('db').changes({
-      since: 'now',
-      live: true,
-      returnDocs: false
-    }).on('change', bind(this, 'onChange'));
+  _hookObservers : on('init', function()  {
+    this.addObserver('db', this, this._startChangesToStoreListener);
```

Comment:
The way this is implemented, the change listener on the old database is never cleared. That means that changes that happen in it will result in attempted updates to the store, even after you've switched databases.

It also means that one database could get the listener applied to it several times (if you switch from database A to B and back to A), which seems like it could cause performance problems if nothing else.

And it means that the change listeners for any database other than the last one set are not torn down when the application is torn down. This will likely cause problems with integration tests.